### PR TITLE
Optimize the Code of Component IoT Form

### PR DESCRIPTION
**Pull request by Kinplemelon**

### CODE REVIEW

File: src/types/rule.d.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export interface OutputItemObj {
   args?: {
     payload: string
     topic: string
+    qos: number
```
Comment (ysfscream) [inline src/types/rule.d.ts:25]: type: 0 | 1 | 2

File: src/views/RuleEngine/components/SQLTestDialog.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  }).catch((e) => {
+    testParams.value.output = e
+  })
+
+  if (res) {
+    try {
+      const text = JSON.stringify(res)
+      testParams.value.output = text
+    } catch (e) {
+      console.log(e)
```
Comment (ysfscream) [inline src/views/RuleEngine/components/SQLTestDialog.vue:117]: message.error(e)?

Comment (ysfscream) [reply src/views/RuleEngine/components/SQLTestDialog.vue:117]: If the json content is incorrect, the user should be able to see the error.

**Review by ysfscream — APPROVED** [review]

File: src/types/rule.d.ts (hunk 1 above)
Comment (Kinplemelon) [reply src/types/rule.d.ts:25]: 👌🏻

File: src/views/RuleEngine/components/SQLTestDialog.vue (hunk 2 above)
Comment (Kinplemelon) [reply src/views/RuleEngine/components/SQLTestDialog.vue:117]: Here is the previous code, I did not modify the logic; I think if the response is not in JSON format?

**Pull request merged**